Content.Server/Cloning/CloningSystem.cs

```diff
@@ -208,6 +209,16 @@ public bool TryCloning(EntityUid uid, EntityUid bodyToClone, Mind.Mind mind, Clo
             clonePod.UsedBiomass = cloningCost;
             // end of biomass checks
 
+            // Check if they have the uncloneable trait
+            if (TryComp<UncloneableComponent>(bodyToClone, out _))
+            {
+                if (clonePod.ConnectedConsole != null)
+                    _chatSystem.TrySendInGameICMessage(clonePod.ConnectedConsole.Value,
+                        Loc.GetString("cloning-console-uncloneable-trait-error"),
+                        InGameICChatType.Speak, false);
+                return false;
+            }
+
```

Comment:
this isn't an issue with your pr as a whole, but this function is begging to be split into a helper function + a cancellable event.

Reply:
What do you mean?

Reply:
the function ideally shouldn't be checking a bunch of data on various components, rather, it should raise an event that these component can subscribe to and cancel on their own. It's not super important but this code is really cluttered because of all the various component checks.

Reply:
So for this PR does this need to be changed, or will this be saved for a different PR? This PR was only really supposed to add the trait.

Reply:
I will assume it doesn't need to be changed.